src/lsp/cobol_parser/grammar_tokens.mly

```diff
@@ -1399,4 +1399,127 @@
 %token XML_NTEXT                    [@keyword.silenced]
 %token XML_TEXT                     [@keyword.silenced]
 
+(* INTRINSIC FUNCTIONS *)
+%token <Cobol_ptree.generic_intrinsic_identifier> INTRINSIC_FUNC
+
+(* Intrinsic functions that need special care (either because their name is a COBOL keyword or
+   because they have special arguments) *)
+%token BYTE_LENGTH_FUNC                  [@intrinsic.unique "BYTE-LENGTH"]
```

Comment:
You could further simplify these declarations by using the same trick as for keywords (that uses the name of the token and replaces `_` with `-` when the `@keyword` has no payload).

Reply:
(At least for the non-unique)